docs/docsite/rst/developing_modules_general_aci.rst

```diff
@@ -0,0 +1,1851 @@
+
+The test cases for the ACI modules follow a consistent pattern across operations such as Create, Update, Query, Query all, and Delete. For Create, Update, and Delete operations, each test includes:
+
+  + The check mode task to simulate the operation without making changes.
+  + The normal_mode task to perform the actual operation.
+  + The normal mode tasks to verify idempotency, ensuring that running the operation again does not cause changes.
+  + After the operations, the test verifies the output files and asserts that the changes occurred as expected or not changed when re-run in check_mode, confirming the module's correct behavior and idempotency.  
+
+This structured approach ensures thorough testing of the modules' functionality and stability in different modes and repeated executions.
+
+
+5. We begin by adding tasks to post configuration to the APIC. This includes creation of all the classes such as tenant and l3out that were used in the construct_url function in our module.
+
+.. code-block:: yaml
+
+      - name: Add a new tenant
+        aci_tenant:
+          <<: *aci_info 
+          tenant: ansible_tenant
+          description: Ansible tenant
+          state: present
+
+      - name: Add a new L3Out
+        aci_l3out:
+          <<: *aci_info
+          tenant: ansible_tenant
+          name: ansible_l3out
+          description: L3Out for ansible_tenant tenant
+          domain: ansible_dom
+          vrf: ansible_vrf
+          l3protocol: ospf
+          route_control: export
+          state: present
+
+      - name: Add a logical node
+        cisco.aci.aci_l3out_logical_node:
+          <<: *aci_info
+          tenant: ansible_tenant
+          l3out: ansible_l3out
+          logical_node: lNode
+          pod_id: 1
+          node_id: 101
+          router_id: "10.1.0.1"
+          router_id_as_loopback: 'yes'
+          state: present
+
+.. code-block:: text
+
+6. The next section consists of adding tasks for all aspects of our module. We include Ansible's register attribute to save the result of the task. The procedure is as follows:
+
+  #. We include the task for adding aci_l3out_static_routes using state: present with no attribute bfd. It consists of most attributes defined in our module.
+  #. We include the task for adding aci_l3out_static_routes again using state: present with the same attributes used in step 1 to check for idempotency.
+  #. We include the task for adding aci_l3out_static_routes using state: present with the bfd attribute.
+  #. We include the task for querying aci_l3out_static_routes for the new attribute bfd using state: query.
+  #. We include the task for adding a new aci_l3out_static_routes using state: present.
+  #. We include the task to query all aci_l3out_static_routes under the root object: tenant, using state: query.
+  #. We include the task for deleting aci_l3out_static_routes using state: absent.
+
+.. code-block:: yaml
```

Comment:
not sure I follow the task order, thought the order we current use is as follows:

1. setup if required
2. create with min config (check/normal/idempotent)
3. update with full config (check/normal/idempotent)
4. additional update cases
5. additional create cases also used for query all
6. queries
7. errors
8. delete (check/normal/idempotent)
9. cleanup